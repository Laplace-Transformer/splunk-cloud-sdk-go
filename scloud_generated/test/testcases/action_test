action,list-actions
action,get-action,--action-name,act1
