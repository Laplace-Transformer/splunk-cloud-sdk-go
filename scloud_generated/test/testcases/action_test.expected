#testcase: action create-action-email-action --kind email --name scloud_test_action_test --members kk@splunk.com --members aa@splunk.com
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions
REQUEST BODY:{{"kind":"email","name":"scloud_test_action_test","members":["kk@splunk.com","aa@splunk.com"]}}

#testcase: action update-action-email-action-mutable --action-name scloud_test_action_test --subject "newsubject"
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions/scloud_test_action_test
REQUEST BODY:{{"subject":"\"newsubject\""}}

#testcase: action get-action --action-name scloud_test_action_test
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions/scloud_test_action_test
REQUEST BODY:<nil>

#testcase: action get-public-webhook-keys
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/system/action/v1beta2/webhook/keys
REQUEST BODY:<nil>

#testcase: action create-action-webhook-action --kind webhook --name scloud_test_action_test_wh --title newtitle --webhook-payload "{{ .name }} is a {{ .species }}" --webhook-url "https://webhook.site/test"
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions
REQUEST BODY:{{"kind":"webhook","name":"scloud_test_action_test_wh","webhookPayload":"\"{{","webhookUrl":"\"https://webhook.site/test\"","title":"newtitle"}}

#testcase: action update-action-webhook-action-mutable --action-name scloud_test_action_test_wh --title "newtitle"
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions/scloud_test_action_test_wh
REQUEST BODY:{{"title":"\"newtitle\""}}

#testcase: action trigger-action --action-name scloud_test_action_test_wh
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions/scloud_test_action_test_wh
REQUEST BODY:{{"actionMetadata":{}}}

#testcase: action get-action-status --action-name scloud_test_action_test --status-id 1e77d173-dfc9-43df-a7f9-ea053aed6f3a
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions/scloud_test_action_test/status/1e77d173-dfc9-43df-a7f9-ea053aed6f3a
REQUEST BODY:<nil>

#testcase: action  list-actions
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions
REQUEST BODY:<nil>

#testcase: action delete-action --action-name scloud_test_action_test
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions/scloud_test_action_test
REQUEST BODY:<nil>

#testcase: action delete-action --action-name scloud_test_action_test_wh
enable testhook-dryrun
REQUEST URL:https://api.staging.scp.splunk.com:443/scloud/action/v1beta2/actions/scloud_test_action_test_wh
REQUEST BODY:<nil>

